src/Compiler/Checking/NicePrint.fs

```diff
@@ -893,6 +903,10 @@ module PrintTypes =
         // Always prefer 'float' to 'float<1>'
         | TType_app (tc, args, _) when tc.IsMeasureableReprTycon && List.forall (isDimensionless g) args ->
           layoutTypeWithInfoAndPrec denv env prec (reduceTyconRefMeasureableOrProvided g tc args)
+        
+        // Special case for nested array<array<'t>> shape
```

Comment:
This is wrong, and this rule should be removed.  Also there are no tests for jagged arrays - they should be added.

Reply:
Will update next week. Thanks for the review